src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,323 @@
+    public Parser() {
+    };
+
+    private static class DateTimeWrapper {
+        private LocalDate date;
+        private LocalTime time;
+
+        private DateTimeWrapper(LocalDate date, LocalTime time) {
+            this.date = date;
+            this.time = time;
+        }
+    }
+
+    /**
+     * Parses the input and returns the command and its arguments as an array of
+     * Strings.
+     * If there is only one word in the input it returns an array of the input and
+     * an empty string.
+     * @param input the command given by the user.
+     */
+    private static String[] parseInput(String input) {
+        int index = input.indexOf(' ');
+        // Returns the input string as a list of strings.
+        if (index > -1) {
+            return input.split(" ", 2);
+        } else {
+            String[] tempString = { input, "" };
+            return tempString;
+        }
+    }
+
+    /**
+     * Parses file path and returns array of Strings representing each level in the
+     * file path.
+     * @param input path to be parsed.
+     */
+    public static String[] parseFilePath(String input) {
+        int index = input.indexOf('/');
+        // Returns the input string as a list of strings.
+        if (index > -1) {
+            return input.split("/", 2);
+        } else {
+            String[] tempString = { input, "" };
+            return tempString;
+        }
+    }
+
+    /**
+     * Parses an input into an integer. This is used to determine which element in a
+     * list to perform an operation on.
+     * @param input string to be parsed. Must be non-null and a valid integer.
+     * @return The index of the input if input is valid.
+     * @throws InvalidNumberException if input isn't valid.
+     */
+    private static int parseIndex(String input) throws DukeException {
+        try {
+            return Integer.parseInt(input) - 1;
+        } catch (NumberFormatException e) {
+            throw new InvalidNumberException();
+        }
+    }
+
+    /**
+     * Parses the input and returns a DateTimeWrapper. This is a convenience method
+     * for deadlineParser and eventParser that will try to parse the date and time
+     * out of the input.
+     * @param input the input to parse. Must be non-null and a valid date or time or
+     *              both.
+     * @return a DateTimeWrapper that contains the date and time of the input if
+     *         applicable.
+     * @throws IncorrectFormatException if input isn't valid.
+     */
+    private static DateTimeWrapper parseDate(String input) throws IncorrectFormatException {
+        // Input is parsed into two if it contains both a date and time component.
+        String[] parsedInput = parseInput(input);
+        LocalDate date;
+        LocalTime time;
+        String i1 = parsedInput[0].trim();
+        String i2 = parsedInput[1].trim();
+
+        // Matches both halves of the input to the date.
+        if (i1.matches("[0-9]{2}.[0-9]{2}.[0-9]{4}")) {
```

Comment:
the variables names i1 and i2 can be improved to make the code more self-explanatory and meaningful to increase code readability 😄